src/crl.c

```diff
@@ -174,8 +174,10 @@ void FreeCRL(WOLFSSL_CRL* crl, int dynamic)
 #ifdef HAVE_CRL_MONITOR
     if (crl->tid != 0) {
         WOLFSSL_MSG("stopping monitor thread");
-        if (StopMonitor(crl->mfd) == 0)
+        if (StopMonitor(crl->mfd) == 0) {
             pthread_join(crl->tid, NULL);
+            close(crl->mfd);
```

Comment:
I decided to move freeing crl->mfd up to FreeCRL instead of DoMonitor to fix this race condition, note that this assumes that FreeCRL will be used to stop the monitor thread.  If this isn't always true, I can use a mutex instead to fix this.

Reply:
What exactly is the race?

crl->mfd is closed by DoMonitor() before DoMonitor() finishes and is joinable.

StopMonitor() is the only other user of crl->mfd and that is to send the custom shutdown event to the thread so that it finishes.

Reply:
The race was DoMonitor closing crl->mfd while StopMonitor was trying to write to it to signal to DoMonitor that it should exit.  If DoMonitor were to exit its loop for any reason other than StopMonitor signalling it, this race condition would be triggered.

I'm not clear if it's actually possible to trigger this, or if the compiler was just being overeager.  Can anything other than StopMonitor cause DoMonitor to exit its loop?